### ISSUE TITLE
XML Formatter: Xml declaration always declares encoding as utf-16

### ISSUE THREAD

**Issue by jwfxpr** [veler/DevToys#355]
### Description

As per e633a058, when the XML input omits the declaration, so too does the output. However, when the xml declaration is there, the xml output _always_ declares the encoding to be `utf-16`, regardless of the encoding declared in the input, or if the encoding attribute is omitted from the declaration entirely.

Examples:
![image](https://user-images.githubusercontent.com/20788820/153003674-1bef1e49-637f-49c8-a97a-aa042592a694.png)
![image](https://user-images.githubusercontent.com/20788820/153003749-b6d7bfdf-3990-43f7-b8d7-ad86a8c5e50c.png)

C# internally represents all strings as utf-16, which is probably what causes this.

### Steps To Reproduce

Paste the following example XML in the input field of the XML formatter:

```xml
<?xml version="1.0"?>
<Document version="0.1">
<Element>
<Sub-Element />
</Element>
</Document>
```

![image](https://user-images.githubusercontent.com/20788820/153004367-d4f4e8da-9ff0-431a-a9e4-836935f90117.png)

### Expected behavior

I would not expect the actual content of the input to be changed in any way by the XML formatter, aside from whitespace. Even though _technically_ the output field is in utf-16, if I copy text from a document and then paste that text _back_ to the document after formatting, it will retain the original document's encoding, and the altered declaration will be wrong, leading in many cases to an invalid XML error when that document is used.

In other words, if the `encoding=""` attribute is omitted from the declaration in the input, I expect it to be omitted in the output; if it in present in the input, I expect it to declare the same encoding in the output, _regardless_ of the correctness of the encoding that the output field uses.

### DevToys Version

Version 1.0.2.0 | X64 | RELEASE | b972462 | b972462

### Relevant Assets/Logs

![image](https://user-images.githubusercontent.com/20788820/153003674-1bef1e49-637f-49c8-a97a-aa042592a694.png)
![image](https://user-images.githubusercontent.com/20788820/153003749-b6d7bfdf-3990-43f7-b8d7-ad86a8c5e50c.png)
![image](https://user-images.githubusercontent.com/20788820/153004367-d4f4e8da-9ff0-431a-a9e4-836935f90117.png)

**Comment by THELEMITED** [comment]
So you have to manually get your phone's network translator to support ums, and utm, khtml, umt, mt5, mt6, esp, esp32, Llvm curses #noirerhel

**Comment by YSternlicht** [comment]
What do you think should happen if the user puts in an invalid encoding?
Should it just ignore or throw an error?

**Comment by YSternlicht** [comment]
Ok. Created a PR
https://github.com/veler/DevToys/pull/364

(First time so wish me luck ☺️

**Comment by jwfxpr** [comment]
Nice one @YSternlicht! I left a comment over on the PR for you! 😄
